### PR TITLE
Add template NextJS dockerfile + build job

### PR DESCRIPTION
**Pull request by matcor-bodc**
for work relating to 8696kuv0g

### CODE REVIEW

**Comment by thogar-computer** [comment]
Task linked: [CU-8696kuv0g Code containerisation (typescript)](https://app.clickup.com/t/8696kuv0g)

File: .github/workflows/ts-build.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    - 'typescript-demo/**'
+
+jobs:
+  build:
+    env:
+      shouldPushImage: ${{ github.ref == 'refs/head/main' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
```
Comment (leobr29) [inline .github/workflows/ts-build.yml:18]: Just a remark for my own personal culture, what's the point of using QEMU here ?

**Review by leobr29 — DISMISSED** [review]
seems good

File: .github/workflows/ts-build.yml (hunk 1 above)
Comment (matcor-bodc) [reply .github/workflows/ts-build.yml:18]: I don't really know to be honest. It's part of all the examples in Github's documentation, so I assume it is required in setting up the Docker build tools

Comment (MCazaly) [reply .github/workflows/ts-build.yml:18]: I suspect this is getting around Docker in Docker.

File: .github/workflows/ts-build.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
+name: Typescript Build
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    paths:
+    - 'typescript-demo/**'
```
Comment (MCazaly) [inline .github/workflows/ts-build.yml:9]: Should we also add the `typescript-demo/**` filter for the main branch jobs?

File: typescript-demo/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,57 @@
+FROM cgr.dev/chainguard/node AS base
+
```
Comment (MCazaly) [inline typescript-demo/Dockerfile:2]: Can we add some metadata via LABEL here? Similar to [these lines](https://github.com/British-Oceanographic-Data-Centre/amrit-repos/blob/main/example-python/Dockerfile#L3) in the Python example.

File: typescript-demo/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+WORKDIR /app
+
+COPY package.json package-lock.json ./
+RUN npm ci
+
+# Rebuild the source code only when needed
+FROM base AS builder
+WORKDIR /app
+COPY --from=deps /app/node_modules ./node_modules
+COPY . .
```
Comment (MCazaly) [inline typescript-demo/Dockerfile:17]: Forgive my Node ignorance here - is copying _everything_ required? Although I suppose it's not that big a concern if we're not keeping the image afterwards.

**Review by MCazaly — CHANGES_REQUESTED** [review]
Looking good! Just a few minor comments from me.

File: typescript-demo/Dockerfile (hunk 4 above)
Comment (matcor-bodc) [reply typescript-demo/Dockerfile:17]: Probably not, but in theory everything not in dockerignore should be there.

File: typescript-demo/Dockerfile (hunk 3 above)
Comment (matcor-bodc) [reply typescript-demo/Dockerfile:2]: Added

**Pull request merged**